src/test/scala/com/github/tarao/slickjdbc/helper/TraitSingletonBehavior.scala

```diff
@@ -21,6 +21,6 @@ trait TraitSingletonBehavior { self: UnitSpec =>
   def exportingTheTraitMethods[T : Manifest](singleton: Any) {
     singleton shouldBe a [T]
     val parent = implicitly[Manifest[T]].runtimeClass.asInstanceOf[Class[T]]
-    signatures(singleton.getClass) should equal (signatures(parent))
+    signatures(singleton.getClass) subsetOf (signatures(parent)) shouldBe true
```

Comment:
Before scala 2.12

> Such a trait would compile to two class files
>   - the trait interface, a Java interface, and
>   - the implementation class, containing "trait implementation methods"

https://github.com/scala/scala/pull/5003#issue-61213624

`implicitly[Manifest[T]].runtimeClass.asInstanceOf[Class[T]]` is the former one.

---

In Scala-2.12 

> no longer emits trait implementation classes or trait implementation methods.
> instead, concrete methods are simply retained in the interface, as JVM 8 default interface methods (the JVM spec changes in JSR-335 paves the way)

https://github.com/scala/scala/pull/5003#issue-61213624

---

Example, (from http://keens.github.io/slide/scala_2_12_0henkouten/)

```scala
// Trait.scala
trait Foo {
  def foo(): String = "foo"
  def bar(): String
}
```

```
$ scalac-2.11 Trait.scala
$ ls
Foo$class.class  Foo.class  Trait.scala
$ javap Foo
Compiled from "Trait.scala"
public interface Foo {
  public abstract java.lang.String foo();
  public abstract java.lang.String bar();
}
$ javap 'Foo$class'
Compiled from "Trait.scala"
public abstract class Foo$class {
  public static java.lang.String foo(Foo);
  public static void $init$(Foo);
}
```

```
$ scalac-2.12 Trait.scala
$ ls
Foo.class  Trait.scala
$ javap Foo
Compiled from "Trait.scala"
public interface Foo {
  public static java.lang.String foo$(Foo);
  public java.lang.String foo();
  public abstract java.lang.String bar();
  public static void $init$(Foo);
```